### DISCUSSION TITLE
MPC sometimes crashed

### DISCUSSION THREAD

**Discussion by pmj110119** [NVlabs/curobo#98] [Software Q&A] [answered]
I'm using the MPC example `examples/isaac_sim/mpc_example.py`, but I keep getting stuck in weird poses and crashing.

When the target moves faster and farther, the probability of MPC collapse is high. On my own robot (the joint speed is relatively low), the probability of crash is very high. On the franka-robot provided by curobo, the probability is smaller, but there is a chance of it happening.

(crash at 12s of the video)

https://github.com/NVlabs/curobo/assets/40192364/10b25d0e-15fb-4076-b1d1-6d0ee304ff52

What may be the reason for this?

**Comment by pmj110119** [comment]
And `MPC` seems to be less stable than the results of `Motion Generation`, the robot cannot accurately stop at the target pose, and the probability of collision is a little high.


I am very curious, for dynamic collision avoidance, whether the following method can be used: 
1. Firstly use `Motion Generation` to obtain a trajectory, while the robot moves along the trajectory, "esdf" is continuously updated and checks whether a collision occurs on the trajectory.
2. If a collision occurs, a new trajectory is planned by `Motion Generation` again and executed.


Compared with MPC, will this method be more stable? I would be grateful if I could get your advice!

**Comment by balakumar-s** [comment, marked as answer]
From the video, it looks like the gripper has some free joints that can cause unstable dynamics. Can you make these joints fixed? This should make mpc tracking better.
